scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -167,6 +168,9 @@ class AsyncFlowDefRunner(mode: CascadingMode) extends Writer {
       @annotation.tailrec
       def go(id: Long): Unit = messageQueue.take match {
         case Stop => ()
+        case StopFlow(flow) =>
```

Comment:
sorry to be a pain in the ass... I'm worried that cascading can throw here on `flow.stop()` and if it did we have to swallow the exception.

Maybe we should do something like:
```scala

case class StopFlow(flow: Flow[_], result: Promise[Unit]) extends FlowDefAction
//...
case StopFlow(flow, promise) =>
  promise.complete(Try(flow.stop()))
  go(id)

// then change CancellationHandler:
def fromFn(fn: ConcurrentExecutionContext => Future[Unit]): CancellationHandler =
  new CancellationHandler {
    def stop()(implicit cec: ConcurrentExecutionContext): Future[Unit] =
      fn(cec)
    }

// finally:

val cancel = fromFn { implicit cec =>
  val done = Promise[Unit]()
  messageQueue.put(StopFlow(flow, done))
  done.future
}
```

Reply:
makes sense, done